### ISSUE TITLE
Mapping tips to deepest nodes can cause taxonomy to override phylogeny

### ISSUE THREAD

**Issue by snacktavish** [OpenTreeOfLife/treemachine#166]
(This issue overlaps with #158 and #159 and is described in https://github.com/OpenTreeOfLife/opentree/wiki/pseudo-code-supertree#step-2-remap-to-uncontested-higher-taxa, but was coming up in email, so I thought it was easiest added here as well)

As I understand it, in order to maximize overlap between trees, tips are mapped to deepest node in that tree.
Therefore that tree is treated as supporting the monophyly of all taxonomic groups which descend from that node.

i.e. Tree (((A1, B1,)),C1)
with Taxonomy (((A1, A2),(Z1,Z2,)),(B1,B2),C1)
After mapping the tree is treated in synthesis as: ((((A1,A2)(Z1,Z2)),(B1,B2)),C1)
@blackrim @chinchliff (Is this a correct depiction of how mapping to higher taxa works?)
 
However, this causes taxonomy to override phylogeny, even in cases where there is no conflict among trees.
Example 1:
High rank phylogeny P1: (((A1,B1),C1),D1);
Lower rank phylogeny P2: (((A1,B1),(B2,C1)),D1);
Taxonomy: (A1,(B1,B2)),C1,D1);
Naive expected synthesis: (((A1,B1),(B2,C1)),D1); because there is no conflict except with taxonomy
Actual expectation (I think) :  ((A1,(B1,B2),),C1),D1); because B1 in tree P1 is mapped as (B1,B2)

example 2:
High rank phylogeny P1: ((X1,Y1),Z1);
Lower rank phylogeny P2: ((Q1,Y1),Q2)),Z1);
Taxonomy: (X1, (Y1,(Q1,Q2)),Z1)
Naive expected synthesis: ((X1,(Q1,Y1),Q2))),Z1);
Actual expectation (I think) : ((X1,(Y1,(Q1,Q2))),Z1);
As I understand it, tip Y1 in tree P1 would be mapped as (Y1,(Q1,Q2))), and override P2.

This seems to contradict our statements about phylogenies taking precedence over taxonomy.

A potential approach to dealing with this problem is described here
https://github.com/OpenTreeOfLife/opentree/wiki/pseudo-code-supertree#step-2-remap-to-uncontested-higher-taxa
In short - only mapping tips to deeper nodes that no other phylogeny suggests are non-monophyletic.

**Comment by chinchliff** [comment]
Hi Emily, getting back to this issue. So, I think this is a challenge, but not a technical one so much as it is just a problem with ambiguity on inputs. I remembered finally why we started "mapping to deepest exemplar" taxa, which was that if you don't, then you lose tree overlap and that leads to a loss in our ability to identify trees that support relationships, especially deep ones. So far I have seen three alternative logical approaches to this problem. Here they are, along with the advantages/disadvantages:

1. Map the input tips exactly to the specified taxa. Advantage: clarity and exactness. Disadvantage: lack of tree overlap especially at deep levels. Exemplar phylogenies (like the generic phylogeny of plants that forms our backbone for synthesis) lose their utility.

2. Map to deepest exemplar taxon. Advantage: maximize tree overlap, critical for synthesis and for reporting tree compatibility information from web services. Disadvantage: has the potential to introduce "pseudoconflict" as shown in above examples--this is only "pseudo" when the intention of the researcher was *not* for the tip to serve as an exemplar of a known (and presumed monophyletic) taxon. If the intention was for it to be an exemplar, then the conflict is real.

3. Map to the deepest uncontested taxon. Advantage: does not introduce pseudoconflict. Disadvantage: high potential for inaccurate representation--the taxon that is presumed to be exemplified by the tip is dependent on the other trees that are present in the database. If we add a supermatrix tree with many phylogenetic errors that breaks the monophyly of many taxa, we lose overlap in our deep trees. Mapping to a deepest uncontested taxon presumes that the intent of the researcher who labeled the tip in question was to map to that taxon as well, but this is probably not going to be true very often.

I propose that the solution here would be to encourage (and provide a mechanism for) curators to identify *exemplified* taxa for the tips in their trees, which would entirely defuse the issues of ambiguity here. In the absence of that clarifying info, I prefer to continue to assume approach 2 as the default, because it (1) has the useful property of having a deterministic outcome that depends only on the tree and the taxonomy and (2) seems to offer the most advantages (maximizing tree overlap) vs. costs (pseudoconflict when we incorrectly assume exemplar status, which i don't think is very often).

**Comment by mtholder** [comment]
What does the intent of the researcher (either the researcher conducting the analysis or the curator) have to do with what the tree is saying? Their intention does not matter anywhere in the analytical pipeline that produced the tree, so it is hard to see why this should have a bearing on our interpretation of the tree.

If the data for the tip was created by making a chimera of several exemplars of a group, then - sure, that mapping of the tip to a higher-than-species taxon is a curation issue. Similarly, if all that is known about the tip is a genus name, then it makes sense for the curated tip to be mapped to that higher level taxon.

But if the data from the tip was not obtained from multiple species, I don't see how intent comes into it.


I think that we are using intent as a rough proxy for confidence of the researcher in the monophyly of taxa. For instance, if you are quite confident in the monophyly of Primates, then you might feel like having one representative for that entire group seems enough.  But I think that the open tree project needs to be capturing that sort of information by getting more trees that support the monophyly of the groups rather than trying to capture curator's view of the intent of the researchers who built the tree.

**Comment by chinchliff** [comment]
But isnt the intent of the researcher what the curation pipeline is all
about? Where is the ingroup, what is the focal clade, in the case of
duplicate tips, which is the one to be used, what is the proper taxon for
this tip, etc. those are all questions of intent. We have assumed so far
that the intent of tips is to represent the deepest taxon they exemplify,
but that does have problems. If we could clarify this intent, we could
avoid the problem of incorrectly assuming what the tree is intended to
represent.

It's easy enough to find the deepest uncontested taxon. So that can be
done, and we can see what happen. But most taxa have broken monophyly in
some tree. If a researcher has a tree A with (((soybean, corn),
yellow pine), sword fern) and in some iteration of the synthesis we say A
is talking about the relationships among (((eudicots, monocots),
gymnosperms), ferns), but then we add more trees (perhaps large super trees
with errors) to the next iteration that contest the monophyly of these
groups, and we then say that A is talking about (((papilionoids, grasses),
pines), polypodiaceae), does not that seem inconsistent?

I agree that the tree itself could speak to both situations. What I'm not
sure about it is how best to handle the ambiguity. The different
tip assignments have different consequences for synthesis, unless we
proliferate relationships for all tips to every appropriate ancestral node
(or carry them back during synthesis). Maybe that's what we have to do to
get around this issue?

**Comment by mtholder** [comment]
I wouldn't say that curation is about capturing intent. I'd say that it is about capturing information that is not available in the raw inputs.

Admittedly there are grey areas, where things aren't clear and we have to rely on expert knowledge. But ideally the things that we try to prompt from the curator should be:
  1. facts which could be meaningfully debated by other experts, and
  2. statements about the study.


  * Presumably the tip mapping is about specifying the taxon in OTT that corresponds to the data that underlies the OTU in the study.
  * I agree that "focal clade" business could be interpreted as the intent of the study - but frankly that is one of the reasons that I've always found that field as basically useless.
  * I think that we use the ingroup designation as a proxy for "we are pretty sure that this subtree is rooted correctly." It is ugly that that statement is not tied to evidence, but the use of the outgroup criterion is so widespread in systematics that it is hard to see how we avoid it.


To be clear, I certainly don't object to have a curated list of statements about how to map "curated tips" to deeper taxa that they exemplify.

I don't even object to that info getting served up and coordinated by the phylesystem-api.

I just think that that info is mainly a set of statements about how we think that tips should be mapped to produce a reliable supertree. It is hard to think about the correct mapping without understanding what supertree operation is going to be performed. So it is really not "study data" and I don't think that it particularly related to the intent of the original researcher.

**Comment by snacktavish** [comment]
I'm still not really getting the issues with option 3).
A synthetic tree is an output of a group of trees, so I don't think that it is inconsistent to interpret relationships in the context of that set of trees.

I don't agree with letting an interpretation of a curators belief in taxonomy overrule phylogenetic statements about relationships. If we have  trees that break up the monophyly of a group, and no trees that support monophyly, why would that group be monophyletic in the synthetic tree?

I didn't understand this part -  "unless we proliferate relationships for all tips to every appropriate ancestral node (or carry them back during synthesis)" Can you explain?

**Comment by chinchliff** [comment]
Ok. Just to be clear, I'm not particularly attached to a specific outcome
here. I agree option 2 has problems. I just think think they're lesser
problems than 3. I am willing to be convinced that they're not.

The part about proliferating relationships means that we would just map a
tip to whatever ott id it comes in with, and also map it to all the deeper
taxa that could be exemplified by that tip given the rest of the taxa in
the input tree (so every taxon between itself and its deepest exemplified
taxon). That could potentially solve this problem, I think--all the
potential exemplar taxa would be identified and synthesis could pick the
correct one given the other inputs. And it might not be the worst thing to
do. For the most part we'd just be introducing a lot of singleton
relationships which I don't think will affect synthesis very much.

Maybe I'm misunderstanding the idea for the "deepest uncontested taxon".
But it seems that whether a taxon X is supported by any tree, it doesn't
matter. X is contested if there is *any* tree that breaks monophyly,
correct? So 99 of 100 trees could support monophyly X, but if one did not
then it would be contested. This seems problematic to me, because it's very
easy to introduce trees that break monophyly. And if we don't map tips to
deep taxa if those taxa are not monophyletic (or unsampled) across *all*
input trees, that seems like a very strong constraint. Again, it only
matters because if we don't map to deep taxa, then we don't get tree
overlap. And again, this may already be soluble by a different approach
entirely... Which would mean my original response to this might have been
premature. So, sorry if that's the case! Still an interesting question. I
guess at this point I'm leaning toward the "proliferation" of relationships
if that's tractable... It seems like it could be and it seems like it would
solve the overall problem of having to decide in advance...

**Comment by snacktavish** [comment]
Coming back to this issue, how does mapping to deepest taxon play into synthesis currently, and what are the expectations for the trees in my first comment? I'm still not sure what behavior would be considered correct according to our criteria. Thanks!

**Comment by blackrim** [comment]
Sorry I haven't closed this earlier because I made it so that you don't have to map deepest. You give it the shallow taxa from the original set and it will from the shallowest to the deepest and everywhere in between. That way, you also don't have to check before hand for what taxa are monophyletic before running the analysis. So your example runs as expected when you don't map deepest.
![graph](https://cloud.githubusercontent.com/assets/160553/6731093/89aa338a-ce18-11e4-8236-0a9846782004.png)
 That stuff is on this commit 21319191be47c66573ab0ad81f5200655bbf6f0a.
